Mathlib/NumberTheory/MahlerMeasure.lean

```diff
@@ -108,4 +113,20 @@ theorem card_mahlerMeasure_le_prod (n : ℕ) (B : ℝ≥0) :
 
 end Northcott
 
+section Cyclotomic
+
+/-- The Mahler measure of a cyclotomic polynomial is 1. -/
+theorem cyclotomic_mahlerMeasure_eq_one {R : Type*} [CommRing R] [Algebra R ℂ] (n : ℕ) :
+    ((cyclotomic n R).map (algebraMap R ℂ)).mahlerMeasure = 1 := by
+  rcases eq_or_ne n 0 with hn | hn
+  · simp [hn]
+  simp only [map_cyclotomic, mahlerMeasure_eq_leadingCoeff_mul_prod_roots, cyclotomic.monic n ℂ,
+    Monic.leadingCoeff, one_mem, CStarRing.norm_of_mem_unitary, one_mul]
+  apply Multiset.prod_eq_one
+  have : NeZero (n : ℂ) := @NeZero.charZero _ _ {out := hn} _ _
```

Comment:
I think it is enough to provide `NeZero n`, and then the typeclass system will find `have : NeZero (n : ℂ)`.